tests/__init__.py

```diff
@@ -1,4 +1,20 @@
 import os
+import unittest
+
+from tests.emulators import GoogleCloudStorageEmulator
 
 
 TESTS_DIR = os.path.dirname(__file__)
+storage_emulator = GoogleCloudStorageEmulator()
+
+
+def startTestRun(instance):
+    storage_emulator.start()
+
+
+def stopTestRun(instance):
+    storage_emulator.stop()
+
```

Comment:
I assume this is to avoid creation of a test server on each setup/teardown?

What will happen if the tests fail/crash? will it leak?

Reply:
Yes and to avoid tests stopping the emulator that they have in common and making other tests fail. I think this will always be run regardless of the state of the tests?

Reply:
I've just checked - `stopTestRun` is called within a `finally` block in `unittest.runner`:

```
try:
    test(result)
finally:
    stopTestRun = getattr(result, 'stopTestRun', None)
    if stopTestRun is not None:
        stopTestRun()
```
so it should run after every test if python itself doesn't fail.